File: 001_two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_i = {}
        for i in range(len(nums)):
            num_i[nums[i]] = i
        for k in range(len(nums)):
            rem = target-nums[k]
            if rem in num_i.keys() and k != num_i[rem]:
                return [k, num_i[rem]]

File: 001_two_sum.java
import java.util.HashMap;
import java.lang.Math;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map  = new HashMap<>();
        int[] result = new int[2]; 
        // for(int i=0; i<nums.length; i++){
        //     map.put(nums[i], i);
        // }

        for(int i=0; i<nums.length; i++){
            int rem = target-nums[i];
            if(map.containsKey(rem) && map.get(rem) != i){
                result[0] = map.get(rem);
                result[1] = i;
                break;
            }
            map.put(nums[i], i);
        }

        return result;
    }
}